src/lib.rs

```diff
@@ -540,12 +539,19 @@ impl<F> WebSocket<F>
     }
 
     /// Queue an outgoing connection on this WebSocket. This method may be called multiple times,
-    /// but the actuall connections will not be established until after `run` is called.
+    /// but the actual connections will not be established until after `run` is called.
     pub fn connect(&mut self, url: url::Url) -> Result<&mut WebSocket<F>> {
+        try!(self.connect_token(url));
+        Ok(self)
+    }
+
+    /// Queues an outgoing connection on this WebSocket and returns the token of the new connection.
+    /// This method may be called multiple times, but the actual connections will not be established
+    /// until after `run` is called.
+    pub fn connect_token(&mut self, url: url::Url) -> Result<Token> {
```

Comment:
The name of the new method is of course open to discussion. I chose the first name that came to my mind.